### PR TITLE
Fix return for translation to typed language

### PR DESCRIPTION
**Pull request by haya14busa**
return FUNC where FUNC returns nothing cannot be translated into
languages which doens't support this functionality (e.g. Go).

### CODE REVIEW

File: autoload/vimlparser.vim [hunk 1]
```diff
@@ -3914,51 +3919,51 @@ function! s:Compiler.compile(node)
   if a:node.type == s:NODE_TOPLEVEL
     return self.compile_toplevel(a:node)
   elseif a:node.type == s:NODE_COMMENT
-    return self.compile_comment(a:node)
+    call self.compile_comment(a:node)
```
Comment (mattn) [inline autoload/vimlparser.vim]: I wonder this change may go wrong. Vim-script  parse each lines even though the lines are not passing. So early return makes good performance result.

Comment (haya14busa) [inline autoload/vimlparser.vim]: Compiler.compile is only used for compiling Vim script AST into S-like-expression.

It's basically useful just for debugging vimlparser and not often called by other tools, so I think it's not performance sensitive code.

I added a commit which add "return s:NIL" for every non-return condition, but maybe we can revert the commit?

Comment (mattn) [inline autoload/vimlparser.vim]:
```vim
if cond1
  "
elseif cond2
elseif cond3
elseif condN
endif
```
This is an issue of internal behavior of Vim. When not return early, Vim script must skip `if` conditions until `endfunction`.

Comment (haya14busa) [inline autoload/vimlparser.vim]: Yes. I understand the problem, but i said this function is not performance sensitive functions compared with parser functions.

And I thought there may be readability problem with lots of "return s:NIL".

Comment (haya14busa) [inline autoload/vimlparser.vim]: but i think it's ok. it's up to you to decide. I already add return statements for every conditions.

**Comment by mattn** [comment]
LGTM, Thank you

**Comment by haya14busa** [comment]
Thanks!

**Pull request merged**